deps/verifier/src/tdx/eventlog.rs

```diff
@@ -67,21 +85,73 @@ impl CcEventLog {
     }
 
     pub fn query_digest(&self, entity: MeasuredEntity) -> Option<String> {
-        let event_desc_prefix = Self::generate_query_key_prefix(entity)?;
-
         for event_entry in self.cc_events.log.clone() {
-            if event_entry.event_desc.len() < event_desc_prefix.len() {
-                continue;
-            }
-            if &event_entry.event_desc[..event_desc_prefix.len()] == event_desc_prefix.as_slice() {
-                let digest = &event_entry.digests[0].digest;
-                return Some(hex::encode(digest));
+            match (entity, event_entry.event_type) {
+                (MeasuredEntity::TdvfKernel, ev)
+                    if ev.as_str() == "EV_EFI_BOOT_SERVICES_APPLICATION" =>
+                {
+                    let raw_bytes = &event_entry.event_desc[KERNEL_VENMEDIA_DEVPATH_OFFSET
+                        ..KERNEL_VENMEDIA_DEVPATH_OFFSET + 2 * entity.to_string().len()];
+
+                    match read_string(raw_bytes) {
+                        Ok(kernel) if kernel == entity.to_string() => {
+                            return Some(hex::encode(&event_entry.digests[0].digest));
+                        }
+                        Ok(s) => warn!("Unknown Vendor Media Device Path: {s}"),
```

Comment:
This pattern with two `Ok` blocks looks a little odd to me. It might be simplest to first read the string from raw_bytes (and error if it fails) and then check if the string matches the entity.

Reply:
Dropped the other `Ok` (for both kernel and event).